### PR TITLE
Purchases: Redesigned pre-cancellation survey for plans

### LINKED ISSUE #55569: Atomic Downgrade: Ask for feedback during plan cancellation/removal dialog in one single step
In order to provide users the ability to cancel/remove a plan from an Atomic site without HE intervention, we need to add some extra steps to the current plan cancellation/removal dialog (e.g. loss content warning https://github.com/Automattic/wp-calypso/issues/55239, or backup download https://github.com/Automattic/wp-calypso/issues/55240).

However, that could lead to a very long dialog with many steps. Let's avoid that and redesign a simpler dialog by combining all the feedback steps into one single step. This is also an opportunity to get a plan cancellation/removal flow with a UI more aligned with recent Calypso redesigns such as the onboarding flow.

Current dialog | Proposed dialog
--- | ---
 | 

Before shipping this redesign to production, we'd need to A/B test it and make sure it does not affect the current stats monitoring plan cancellations/removals.

**Links**
- Figma file: hDzpuFGlCMGsFxfyFJamJ0-fi-32%3A174
- Experiment design: pbxNRc-Ue-p2

### PR DESCRIPTION
**Pull request by mmtr**
Fixes https://github.com/Automattic/wp-calypso/issues/55569

### Changes proposed in this Pull Request

* Introduces a new `BlankCanvas` component to be used across Calypso when there is a need of rendering a simple and clean layout (which imitates the Gutenberg style of the new onboarding flow).
* Replaces the pre-cancellation survey with a fullscreen dialog (which uses the new `BlankCanvas` component) when cancelling/removing a plan and the `purchases/cancel-plan-fullscreen-form` feature flag is active.

Desktop | Mobile
--- | ---
Screen Shot 2021-09-01 at 11 12 55 | Screen Shot 2021-09-01 at 11 34 35

#### Plan removal (feature flag disabled)
- Use the Calypso live link below.
- Switch to a Simple site without plan.
- Go to Upgrades > Plans.
- Purchase a paid plan with credits.
- Go to Upgrades > Purchases.
- Try removing the plan you just bought. 
- Make sure you get the same pre-cancellation survey as in prod.
- Make sure you can remove the plan successfully.

#### Plan removal (feature flag enabled)
- Use the Calypso live link below.
- Append `?flags=purchases/cancel-plan-fullscreen-form` at the end of the URL to enable the feature flag.
- Switch to a Simple site without plan.
- Go to Upgrades > Plans.
- Purchase a paid plan with credits.
- Go to Upgrades > Purchases.
- Try removing the plan you just bought. 
- Make sure you get the redesigned pre-cancellation survey.
- Make sure the UI matches the design of the Figma file for both mobile and desktop viewports: hDzpuFGlCMGsFxfyFJamJ0-fi-32%3A174.
- Make sure the UI refers to this flow as "Remove plan".
- Make sure you can remove the plan successfully.

#### Plan cancellation (feature flag disabled)
- Use the Calypso live link below.
- Switch to a Simple site without plan.
- Go to Upgrades > Plans.
- Purchase a paid plan with a credit card (you can use the sandbox store: PCYsg-lW4-p2#sandbox-method).
- Go to Upgrades > Purchases.
- Try cancelling the plan you just bought. 
- Make sure you get the same pre-cancellation survey as in prod.
- Make sure you can cancel the plan successfully.

#### Plan cancellation (feature flag enabled)
- Use the Calypso live link below.
- Append `?flags=purchases/cancel-plan-fullscreen-form` at the end of the URL to enable the feature flag.
- Switch to a Simple site without plan.
- Go to Upgrades > Plans.
- Purchase a paid plan with a credit card (you can use the sandbox store: PCYsg-lW4-p2#sandbox-method).
- Go to Upgrades > Purchases.
- Try cancelling the plan you just bought. 
- Make sure you get the redesigned pre-cancellation survey.
- Make sure the UI matches the design of the Figma file for both mobile and desktop viewports: hDzpuFGlCMGsFxfyFJamJ0-fi-32%3A174.
- Make sure the UI refers to this flow as "Cancel plan".
- Make sure you can cancel the plan successfully.

#### Happychat
- Connect to the staging Happychat HUD (PCYsg-b99-p2).
- Follow instructions above to cancel/remove a plan with the feature flag enabled.
- Make sure you see the "Need help? Chat with us" button when you're marked as available in the HUD.
- Make sure you don't see the "Need help? Chat with us" button when you're marked as unavailable in the HUD, or when you're not connected to the the HUD.

#### Upsells
- Follow instructions above to cancel/remove a plan with the feature flag enabled.
- Make sure you get the upsell below when removing/cancelling a Business plan and after selecting "I couldn't install a plugin/theme I wanted". 
  - Screen Shot 2021-09-01 at 11 25 44
- Make sure you get the upsell below when removing/cancelling a Premium plan and after selecting "I couldn't install a plugin/theme I wanted".
  - Screen Shot 2021-09-01 at 11 24 53
- Make sure you get the upsell below when **cancelling** a Premium plan and after selecting "I couldn't install a plugin/theme I wanted".
  - Screen Shot 2021-09-01 at 11 30 45

### CODE REVIEW

**Comment by BogdanUngureanu** [comment]
Great job, @mmtr, LGTM! @dsas can you also have a look at it? :)

**Comment by dsas** [comment]
Overall this is looking great 👍 

Perhaps the need to expand on "Another reason" could be a little more obvious? You have to provide some text before the Cancel plan button becomes enabled. Some explanatory text/placeholder text might help, or a change to the "Another reason" wording.
image

There's a little moment after pressing "Cancel plan", where it changes to "cancelling" for a split second and then reverts back to "Cancel plan" while the cancelling is happening until it finishes. See https://d.pr/v/YiseMq in case a video helps. This doesn't happen when using the personal upsell when cancelling a premium plan.


### Possibly out of scope but for completeness
Not sure about the upsell when cancelling a business plan and choosing "I couldn't install a plugin/theme I wanted". It says "New! Install Custom Plugins and Themes" and also in the paragraph says "Are you sure....lose access to these new features". But what does new mean here? Atomic sites have been a feature for a while, we know the customer is within a refund period but we don't check whether this is a renewal (guessing) or whether the customer has other sites with plugins installed.

On a business plan, when I choose "The plan was too expensive", the second question is "How can we improve our upgrades" which I'm not sure necessarily follows. Should it be something like "How can we improve the value of the plan?", or at least use "plan" rather than "upgrades" in the initial question.

**Comment by mmtr** [comment]
> Perhaps the need to expand on "Another reason" could be a little more obvious? 

Good idea! Done in 0b60175e1ffc9c825565c7850b16a413fccd3c12.

Screen Shot 2021-09-02 at 16 35 06

> reverts back to "Cancel plan" while the cancelling is happening until it finishes

Nice catch. Fixed in c9fa7b44c54630066d650852fa3a3089468db25c

https://user-images.githubusercontent.com/1233880/131865921-27aa3b54-b013-4843-bc63-f9669705f6a6.mov

> It says "New! Install Custom Plugins and Themes"

FWIW, that's the existing wording currently used in the old survey as well. Since we're reusing the same component for rendering the content of the upsells:

Screen Shot 2021-08-24 at 15 17 09

I agree the wording can be improved, but given it's not been introduced in this PR, I think we can leave it for a follow up.

> we know the customer is within a refund period but we don't check whether this is a renewal (guessing)

We actually do, this upsell is also displayed to users who are removing the plan (so they are out the refund period and don't want to renew).

> whether the customer has other sites with plugins installed.

We also check this, the upsell is only displayed when removing/cancelling a plan from a Simple site which doesn't allow to install plugins.

> when I choose "The plan was too expensive", the second question is "How can we improve our upgrades"

Another existing behavior which I decided to maintain in order to keep the scope as narrowed as possible:

Screen Shot 2021-09-02 at 16 32 05

File: client/components/marketing-survey/cancel-purchase-form/index.jsx [hunk 1]
```diff
@@ -313,10 +365,87 @@ class CancelPurchaseForm extends React.Component {
...
+					</Upsell>
+				);
+			case 'upgrade-atomic':
+				return (
+					<Upsell
+						actionHref={ `/checkout/${ selectedSite.slug }/business?coupon=BIZC25` }
+						actionOnClick={ () =>
+							this.props.recordTracksEvent( 'calypso_cancellation_upgrade_at_step_upgrade_click' )
+						}
+						actionText={ translate( 'Upgrade my site' ) }
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-purchase-form/index.jsx:415]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **25** times:
`translate( 'Upgrade My Site' )` *ES Score: 9*

File: client/components/marketing-survey/cancel-purchase-form/index.jsx [hunk 2]
```diff
@@ -333,8 +462,8 @@ class CancelPurchaseForm extends React.Component {
 			.filter( ( product ) => product.term === TERM_ANNUALLY )
 			.map( ( product ) => ( {
 				value: product.productSlug,
-				label: translate( 'Jetpack {{planName/}}', {
-					components: { planName: <>{ product.shortName }</> },
+				label: translate( 'Jetpack %(planName)s', {
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-purchase-form/index.jsx:465]: :warning: This change will be queued for retranslation. We'll display the English text until then.

File: client/components/marketing-survey/cancel-purchase-form/index.jsx [hunk 3]
```diff
@@ -577,7 +820,29 @@ class CancelPurchaseForm extends React.Component {
...
+		if ( surveyStep === steps.FEEDBACK_STEP ) {
+			if ( ! isSurveyFilledIn( this.state, isImport ) ) {
+				// Do not display this question unless user has already answered previous questions.
+				return null;
+			}
+
+			return (
+				<div className="cancel-purchase-form__feedback-question">
+					<TextareaControl
+						label={ translate( "What's one thing we could have done better?" ) }
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-purchase-form/index.jsx:835]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'What's one thing we could have done better? (optional)' )` *ES Score: 11*

File: client/components/marketing-survey/cancel-purchase-form/index.jsx [hunk 4]
```diff
@@ -778,8 +1024,32 @@ class CancelPurchaseForm extends React.Component {
...
 		const { flowType, translate, disableButtons, purchase, isImport } = this.props;
-		const { surveyStep } = this.state;
-		const disabled = disableButtons || this.state.isSubmitting;
+		const { isSubmitting, surveyStep } = this.state;
+		const disabled = disableButtons || isSubmitting;
+
+		if ( surveyStep === steps.FEEDBACK_STEP ) {
+			let actionText;
+			if ( flowType === CANCEL_FLOW_TYPE.REMOVE ) {
+				actionText = disabled ? translate( 'Removing…' ) : translate( 'Remove plan' );
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-purchase-form/index.jsx:1033]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **24** times:
`translate( 'Remove Plan' )` *ES Score: 10*

File: client/components/marketing-survey/cancel-purchase-form/index.jsx [hunk 5]
```diff
@@ -778,8 +1024,32 @@ class CancelPurchaseForm extends React.Component {
...
+			} else {
+				actionText = disabled ? translate( 'Cancelling…' ) : translate( 'Cancel plan' );
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-purchase-form/index.jsx:1035]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Cancelling' )` *ES Score: 13*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/try/cancel-plan-single-step-survey)

Comment (a8ci18n) [inline client/components/marketing-survey/cancel-purchase-form/index.jsx:1035]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Cancel my plan' )` *ES Score: 7*

File: client/components/marketing-survey/cancel-purchase-form/index.jsx [hunk 6]
```diff
@@ -670,6 +904,29 @@ class CancelPurchaseForm extends React.Component {
 			isJetpackProductSlug( purchase.productSlug ) || isJetpackPlanSlug( purchase.productSlug );
 		const productName = isJetpack ? translate( 'Jetpack' ) : translate( 'WordPress.com' );
 
+		if ( surveyStep === steps.FEEDBACK_STEP ) {
+			return (
+				<>
+					<FormattedHeader
+						brandFont
+						headerText={ translate( 'Your thoughts are needed' ) }
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-purchase-form/index.jsx:912]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **24** times:
`translate( 'Your thoughts are needed.' )` *ES Score: 12*

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6563038

Thank you @mmtr for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6563038).

**Pull request merged**